android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -296,32 +296,33 @@ public void onListItemClick(ListView l, View v, int position, long id) {
 
   @Override
   public void onUrlDeviceDiscoveryUpdate() {
+    Log.v(TAG, "onUrlDeviceDiscoveryUpdate");
+    if (SwipeDismissListViewTouchListener.isLocked()) {
```

Comment:
I think this check needs to happen on the UI thread, and it's already handled by emptyGroupIdQueue, so just remove it.